extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -176,6 +176,16 @@ public void handle(RoutingContext event) {
             if (allowedMethods != null) {
                 response.headers().add(HttpHeaders.ACCESS_CONTROL_ALLOW_METHODS, allowedMethods);
             }
+
+            //always set expose headers if present
+            if (exposedHeaders != null) {
```

Comment:
@rhuan080 Is this block redundant, as you set this header below if it is not configured as a wildcard ?

Reply:
@sberyozkin No, it is not. The exposedHeaders variable is the ACCESS_CONTROL_REQUEST_HEADERS coming from request, and the value is added. If the corsConfig.exposedHeaders has a value different from the wildcard, the corsConfig.exposedHeaders is used.

Reply:
@rhuan080 

Why do the request headers have to be reported as exposed headers ?

FYI, the constructor has this line:
```
this.exposedHeaders = createHeaderString(this.corsConfig.exposedHeaders);
``` 

So it does look like to me it is a redundant block, am I still missing something ?

Reply:
Hi, the wildcard allows all headers to be exposed, thus if the client sends the ACCESS_CONTROL_REQUEST_HEADERS it should be returned at Access-Control-Expose-Headers. But, if wildcard is not used and you use something like that (x-custom-header,y-custom-header) the Access-Control-Expose-Headers should not be returned, but the x-custom-header,y-custom-header configured.

Reply:
Note that it is as the class was written. I'm just updating the Access-Control-Expose-Headers to be returned to no-prefight requests.

Reply:
@rhuan080 Hi, I'm sorry if I'm a bit slow today, but I just don't follow this change.

>if the client sends the ACCESS_CONTROL_REQUEST_HEADERS it should be returned at Access-Control-Expose-Headers

Can you point me to the relevant spec text requiring it please ? As far as ACCESS_CONTROL_REQUEST_HEADERS is concerned, my understanding it is about the browser allowing certain headers be sent to the target by XHR/Fetch, while exposed headers are about telling to the browser which extra headers can be available in the response.

>Note that the class remains unchanged; I am solely updating the Access-Control-Expose-Headers to be returned for non-preflight requests.

Can you check my comment about your PR using essentially the same property twice ?

Reply:
Hi @sberyozkin, I think you are right, but maybe it is another issue. Today, the Quarkus set the ACCESS_CONTROL_REQUEST_HEADERS to Access-Control-Expose-Headers when the wildcard (*) is used. However, it should set the response headers returned. Does that make sense? 

We can work on it on another issue. What do you think?

Reply:
Hi, I agree! I will solve this

Reply:
Thanks @rhuan080, it looks better,  but I thought it should be the bock you removed 
Which should stay, i.e, add the exposed headers only if they are non wildcard.

Can you please restore that one and remove the current one, and also, do the wikcard check in the constructor, similarly to how it is done for origins ? That would avoid doing this check on every request.

Nearly there, thanks 👍

Reply:
I guess responding with a wildcard won't be a problem but it won't be of any help to the browser either ?

Reply:
Hi @sberyozkin,

The exposedHeaders variable has a calculated value from the createHeaderString method. This method already considers the wildcard ( return null to wildcard). Thus, I conclude that calling the isConfiguredWithWildcard method is unnecessary.